contracts/ExitFormat.sol

```diff
@@ -137,8 +143,18 @@ library ExitFormat {
                 (bool success, ) = destination.call{value: amount}(""); //solhint-disable-line avoid-low-level-calls
                 require(success, "Could not transfer ETH");
             } else {
-                // TODO support other token types via the singleAssetExit.metadata field
-                ERC20Interface(asset).transfer(destination, amount);
+                // Inspect the metadata to see what kind of token to transfer
+                bytes memory _metadata_ = singleAssetExit.metadata;
+                // Empty metadata or 0 => ERC20
+                if( _metadata_.length == 0 || _metadata_[31] == bytes1(uint8(ExitMetadataType.ERC20)) ) {
+                    IERC20(asset).transfer(destination, amount);
+                }
+                // 1 => ERC1155 
+                else if (_metadata_[31] == bytes1(uint8(ExitMetadataType.ERC1155)) ) {
+                    (,ERC1155ExitMetadata memory metadata) = abi.decode(_metadata_, (uint8, ERC1155ExitMetadata));
+                    // the metadata from the allocation is passed to the safeTransferFrom call
```

Comment:
Curious to know if there is an anticipated use for this metadata?

Reply:
None at this point. I'm not sure how other receiving contracts are making use of this but it seems smart to support it. I suppose in theory a contract might not accept a transfer unless it comes with the appropriate data payload.